src/module.ts

```diff
@@ -69,20 +71,20 @@ export default defineNuxtModule<ModuleOptions>({
     nuxt.options.build.transpile.push(runtimeDir)
 
     if (!options.apiKey) {
-      throw new Error('`[@nuxtjs/algolia]` Missing `apiKey`')
+      logger.warn('Missing `apiKey`')
```

Comment:
issue: The problem with this approach is that it will certainly fix the issue reported in the linked issue however it will break the desired functionality. 

The initial idea was to throw errors here because users should be allowed to run the app if they have not added `apiKey` or `applicationId`. There is no point in running the module if these two are not there. So with your change it will allow users to run the app even when they wont pass required variables.

What I would recommend to do here is maybe to convert these warns into errors and also add a `return` after each of these loggers so that the module won't run and there will be an error in the console showing user what is missing.

Reply:
This is a tricky one. I totally see the reason behind throwing errors, the issue is that it breaks the `prepare` script when the user (or the CI) agent does not have these values set up. I chose to use warn for the same reason: it really is not an error during the prepare script phase, only during runtime.

However I think I found a better solution: [we can check if the module is running during prepare](https://github.com/nuxt-modules/fontaine/blob/main/src/module.ts#L64-L65), and we could remove this checks then. If the module is not running during prepare, we can stick to throwing errors I think.

Reply:
Now if we run `prepare`:

![image](https://github.com/nuxt-modules/algolia/assets/31937175/02212b36-f7db-4913-9286-294edd83ae4d)

And for dev:

![image](https://github.com/nuxt-modules/algolia/assets/31937175/a07742d7-8a72-4fd0-a0fb-389df3b79b5a)

Reply:
This is much better!

Thanks for finding a way to solve the bug and maintain the previous functionality!

I will review it once again and merge it with the upcoming 2.0.0 release :)